### PR TITLE
[CALCITE-6137] Upgrade Gradle from 8.1.1 to 8.5, support jdk21

### PR DESCRIPTION
**Pull request by snuyanzin**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -16,8 +16,8 @@
 #
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=e111cb9948407e26351227dabce49822fb88c37ee72f1d1582a69c68af2e702f
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
+distributionSha256Sum=3e1af3ae886920c3ac87f7a91f816c0c7c436f276a6eefdb3da152100fef72ae
```
Comment (snuyanzin) [inline gradle/wrapper/gradle-wrapper.properties:19]: taken from https://gradle.org/release-checksums/

File: gradlew [hunk 2]
```diff
@@ -85,9 +85,6 @@ done
 APP_BASE_NAME=${0##*/}
 APP_HOME=$( cd "${APP_HOME:-./}" && pwd -P ) || exit
 
-# Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
```
Comment (snuyanzin) [inline gradlew:88]: every change in this file is a result of command `./gradlew  wrapper --gradle-version 8.4`

File: gradle.properties [hunk 3]
```diff
@@ -51,10 +51,12 @@ org.owasp.dependencycheck.version=5.2.2
 # Tools
 checkerframework.version=3.10.0
 checkstyle.version=10.3.2
+jdk8.checkstyle.version=9.3
 spotbugs.version=3.1.11
 
 asm.version=7.1
 bouncycastle.version=1.70
+bytebuddy.version=1.14.10
```
Comment (snuyanzin) [inline gradle.properties:59]: bytebuddy which came with mockito does not support java 21

**Review by stoty — APPROVED** [review]
+1 LGTM

I was going to suggest upgrading Mockito, but then realized that 5.x has dropped Java 8 support.

**Comment by stoty** [comment]
Do you plan to merge this @snuyanzin  ?

**Comment by snuyanzin** [comment]
squashed commits

**Pull request merged**